tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,186 @@
+        specified is outside the image
+        """
+        self.assertRaises(file_system_exceptions.ArgumentsOutOfRangeException,
+                          self.fs.read, CEPH_IMG, INDEX_OUTSIDE_THE_IMAGE,
+                          len(TEST_DATA))
+
+    def test_argument_out_of_range_exception_in_write(self):
+        """
+        This error will be raised in write function if part of the write would
+        fall outside the image
+        """
+        self.assertRaises(file_system_exceptions.ArgumentsOutOfRangeException,
+                          self.fs.write, CEPH_IMG, TEST_DATA,
+                          INDEX_OUTSIDE_THE_IMAGE)
+
+    def test_write_and_read(self):
+        """ Test writting data into image """
+        bytes_written = self.fs.write(CEPH_IMG, TEST_DATA, OFFSET)
+        data_read = self.fs.read(CEPH_IMG, OFFSET, bytes_written)
+        self.assertEqual(TEST_DATA, data_read)
+
+    def tearDown(self):
+        self.fs.remove(CEPH_IMG)
+
+
+class TestSnapshot(unittest.TestCase):
+    """ Test snapshot operations """
+    @trace
+    def setUp(self):
+        self.fs = ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+        self.fs.snap_protect(CEPH_IMG, CEPH_SNAP_IMG)
+
+    def test_list_snapshots(self):
+        """ Test if list_snapshot function lists all snapshots of a image """
+        self.assertIn(CEPH_SNAP_IMG, self.fs.list_snapshots(CEPH_IMG))
+
+    def test_image_exist_exception(self):
+        """ This error will be raised if snap image already exists """
+        self.assertRaises(file_system_exceptions.ImageExistsException,
+                          self.fs.snap_image, CEPH_IMG, CEPH_SNAP_IMG)
+
+    def test_image_has_snapshot_exception(self):
+        """
+        This error will be raised if user removes a image which has snapshot
+        """
+        self.assertRaises(file_system_exceptions.ImageHasSnapshotException,
+                          self.fs.remove, CEPH_IMG)
+
+    def test_snapshot_busy_exception(self):
+        """ This error will be raised if user removes a protected snapshot """
+        self.assertRaises(file_system_exceptions.SnapshotBusyException,
+                          self.fs.remove_snapshot, CEPH_IMG, CEPH_SNAP_IMG)
+
+    def test_snap_protect(self):
+        """
+        Test snap_protect. If a snapshot is marked as protected,
+        function is_snap_protected will return True
+        """
+        isProtected = self.fs.is_snap_protected(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(isProtected, "Failed to protect a snapshot!")
+
+    def test_snap_unprotect(self):
+        """
+        Test snap_unprotect. If a snapshot is marked as unprotected,
+        function is_snap_protected will return False
+        """
+        self.fs.snap_unprotect(CEPH_IMG, CEPH_SNAP_IMG)
+        isProtected = self.fs.is_snap_protected(CEPH_IMG, CEPH_SNAP_IMG)
```

Comment:
It should be is_protected. CamelCase is not used in methods and variables in python.